### PR TITLE
Allow passing the bearer token via query parameter

### PR DESCRIPTION
**Pull request by srueg**
Some applications don't support passing an authorization header. Using
a URL query parameter should always be possible.

Signed-off-by: Simon Rüegg <simon.ruegg@vshn.ch>

### CODE REVIEW

File: webhook/hook.go [hunk 1]
```diff
@@ -42,15 +45,20 @@ func asJSON(w http.ResponseWriter, status int, message string) {
...
+	if tokenHeader != "" {
+		headerElems := strings.Split(tokenHeader, " ")
+		if len(headerElems) != 2 || (len(headerElems) > 0 && headerElems[0] != "Bearer") {
+			return fmt.Errorf("Malformed Authorization header")
+		}
+		token = headerElems[1]
+	} else if tokenQuery != "" {
+		token = tokenQuery
+	} else {
+		return fmt.Errorf("Request does not have bearer token")
```
Comment (corvus-ch) [inline webhook/hook.go]: The _bearer_ might be associated with the Authorization header too much. I would phrase this as:

Request dos not contain an Authorization token

In general I have the impression that we can improve the wording around _Bearer_, _Authorization_, _token_ and the like. Reading through the code is somewhat confusing when it comes to those terms and the context they are used in.

File: webhook/hook.go [hunk 2]
```diff
@@ -42,15 +45,20 @@ func asJSON(w http.ResponseWriter, status int, message string) {
...
 
 func checkBearerToken(r *http.Request, c config.Configuration) error {
-	tokenHeader := r.Header.Get("Authorization")
-	if tokenHeader == "" {
-		return fmt.Errorf("Request does not have Authorization header")
-	}
-	headerElems := strings.Split(tokenHeader, " ")
-	if len(headerElems) != 2 || (len(headerElems) > 0 && headerElems[0] != "Bearer") {
-		return fmt.Errorf("Malformed Authorization header")
+	tokenHeader := r.Header.Get(AuthorizationHeader)
```
Comment (corvus-ch) [inline webhook/hook.go]: A constant got introduced for the header but for the query parameter in the next line, an inline string gets used. I can live with both but would like to have consistency.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,12 @@ Please document all notable changes to this project in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [v0.4.0]
```
Comment (corvus-ch) [inline CHANGELOG.md]: Until there is an actual release, this should be `[Unreleased]`.

Comment (srueg) [reply CHANGELOG.md]: I'd like to cut a release once this is merged.

**Pull request merged**